### ISSUE TITLE
During upgrade from v3 to v4 SQLSTATE Exception while performing migrations

### ISSUE THREAD

**Issue by rob-baker-ar** [solspace/craft-freeform#410]
While upgrading to Craft 4 I am getting an exception during running of migrations:

```
Exception: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'freeform_submissions_enquiry_form_1' already exists
The SQL being executed was: CREATE TABLE `freeform_submissions_enquiry_form_1` (
        `id` int(11) NOT NULL,
        `name_15` text,
        `email_3` text,
        `phone_number_16` text,
        `preferred_contact_17` text,
        `callback_18` text,
        `holiday_19` text,
        `additional_information_50` text
) ENGINE = InnoDb DEFAULT CHARACTER SET = utf8
...
```

Prior to the migration running, there was a `freeform_submissions` table present but none specific to the `enquiryForm` form - note there is no integer id in that form's handle. After this above exception, the table with schema as below was present _(it looks like it tried to create it twice)_.

Result of `SHOW CREATE TABLE freeform_submissions_enquiry_form_1;`:

```
CREATE TABLE `freeform_submissions_enquiry_form_1` (
  `id` int(11) NOT NULL,
  `name_15` text DEFAULT NULL,
  `email_3` text DEFAULT NULL,
  `phone_number_16` text DEFAULT NULL,
  `preferred_contact_17` text DEFAULT NULL,
  `callback_18` text DEFAULT NULL,
  `holiday_19` text DEFAULT NULL,
  `additional_information_50` text DEFAULT NULL,
  PRIMARY KEY (`id`),
  CONSTRAINT `fk_etdiizyonfrqglgrhtomqfohzztrhayqnlcz` FOREIGN KEY (`id`) REFERENCES `freeform_submissions` (`id`) ON DELETE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3
```

**Steps to reproduce**

1. Have a Craft v3.7.42 install
2. Upgrade to Craft v4.1.4.1 by altering versions in `composer.json` and running `composer update` as recommended [here](https://craftcms.com/docs/4.x/upgrade.html#preparing-for-the-upgrade)
3. Install works OK but...
4. run `craft migrate/all`
5. Exception thrown during ` m220330_111857_SplitSubmissionsTable` Freeform migration

**Expected behavior**

Migrations work / no exception.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: `3.7.42` upgrading to `4.1.4.1`
- Freeform Version: `3.13.9` upgrading to `4.0.0-beta.17`
- Freeform Edition: standard
- Fresh Install or Upgrade: Upgrade

**Comment by rob-baker-ar** [comment]
It might be useful to know that these are the relevant migrations that the `craft migrate/all` command had lined up:

```
Total 7 new Freeform migrations to be applied:
    - m220316_060248_SwitchFormSuccessBehaviourToRedirectReturnUrl
    - m220322_070819_RenameFormattingTemplates
    - m220323_113852_MigrateEmailValuesToString
    - m220330_111857_SplitSubmissionsTable
    - m220422_065929_AddExportNotificationsTable
    - m220527_055207_ExpandIntegrationAccessTokenSize
    - m220530_052327_MigrateFormContentTableNamesToSnakeCase
```

**Comment by rob-baker-ar** [comment]
Closing.

I just looked at this again. After the first attempt at updating which failed for unrelated reasons, the DB table in question got added correctly. When it came round to running again: I have a routine that drops database tables in the DEV database before cloning in the live version to try again. This routine was not properly considering tables that got added during migration.

i.e. dropping all tables from dev db and re-dumping them in from live and running migrations again fixed this (as the original issue that broke the migrations first time around is now sorted).
